### PR TITLE
Automatic eigenvalue approximation in ChebConv

### PR DESCRIPTION
**Pull request by daniel-unyi-42**
I modified ChebConv according to the issue [Eigenvalues in ChebConv](https://github.com/pyg-team/pytorch_geometric/issues/3996) and updated the documentation as well.

### CODE REVIEW

File: torch_geometric/nn/conv/cheb_conv.py [hunk 1]
```diff
@@ -113,6 +111,13 @@ def __norm__(self, edge_index, num_nodes: Optional[int],
                                                 normalization, dtype,
                                                 num_nodes)
 
+        if lambda_max is None:
+            lambda_max = 2.0 * edge_weight
```
Comment (rusty1s) [inline torch_geometric/nn/conv/cheb_conv.py:115]: Shouldn't this be?
```python
lambda_max = torch.tensor(2.0, dtype=dtype, device=edge_index.device)
```

File: torch_geometric/nn/conv/cheb_conv.py [hunk 2]
```diff
@@ -113,6 +111,13 @@ def __norm__(self, edge_index, num_nodes: Optional[int],
...
+        if not isinstance(lambda_max, torch.Tensor):
```
Comment (rusty1s) [inline torch_geometric/nn/conv/cheb_conv.py:116]:
```suggestion
        elif not isinstance(lambda_max, torch.Tensor):
```

File: torch_geometric/nn/conv/cheb_conv.py [hunk 3]
```diff
@@ -113,6 +111,13 @@ def __norm__(self, edge_index, num_nodes: Optional[int],
...
+            lambda_max = torch.tensor(lambda_max, dtype=edge_weight.dtype,
```
Comment (rusty1s) [inline torch_geometric/nn/conv/cheb_conv.py:117]:
```suggestion
            lambda_max = torch.tensor(lambda_max, dtype=dtype,
```

File: torch_geometric/nn/conv/cheb_conv.py [hunk 4]
```diff
@@ -113,6 +111,13 @@ def __norm__(self, edge_index, num_nodes: Optional[int],
...
+                                      device=edge_weight.device)
```
Comment (rusty1s) [inline torch_geometric/nn/conv/cheb_conv.py:118]:
```suggestion
                                      device=edge_index.device)
```

File: torch_geometric/nn/conv/cheb_conv.py (hunk 1 above)
Comment (daniel-unyi-42) [reply torch_geometric/nn/conv/cheb_conv.py:115]: ### Sorry, I made a mistake. It should be:
`lambda_max = 2.0 * edge_weight.max()`
If normalization is 'sym' or 'rw' then `edge_weight.max()` equals 1.

File: torch_geometric/nn/conv/cheb_conv.py (hunk 2 above)
Comment (daniel-unyi-42) [reply torch_geometric/nn/conv/cheb_conv.py:116]: I preserved this if statement because the original code included it.
But since `lambda_max` is an OptTensor, it is either `None` or an instance of `torch.Tensor`.
So I think it could be deleted.

**Review by rusty1s — APPROVED** [review]
Thank you!
